cli/azd/internal/repository/app_init.go

```diff
@@ -791,6 +799,68 @@ func (i *Initializer) prjConfigFromDetect(
 	return config, nil
 }
 
+func checkPasswordlessConfigurationAndContinueProvision(database appdetect.DatabaseDep, authType internal.AuthType,
+	detect *detectConfirm, console input.Console, ctx context.Context) (bool, error) {
+	if authType != internal.AuthTypeUserAssignedManagedIdentity {
+		return true, nil
+	}
+	for i, prj := range detect.Services {
+		if prj.Language == appdetect.Java &&
+			(!prj.Metadata.ContainsDependencySpringCloudAzureStarter ||
+				prj.Metadata.ContainsPropertySpringDatasourcePassword) {
+			message := fmt.Sprintf("You selected %s as auth type for %s.",
+				internal.AuthTypeUserAssignedManagedIdentity, database)
+			if database == appdetect.DbPostgres && !prj.Metadata.ContainsDependencySpringCloudAzureStarterJdbcPostgresql {
+				message = fmt.Sprintf("%s This dependency is required: "+
+					"'com.azure.spring:spring-cloud-azure-starter-jdbc-postgresql'. "+
+					"But this dependency is not found in your project: %s.", message, prj.Path)
+			}
+			if database == appdetect.DbMySql && !prj.Metadata.ContainsDependencySpringCloudAzureStarterJdbcMysql {
+				message = fmt.Sprintf("%s This dependency is required: "+
+					"'com.azure.spring:spring-cloud-azure-starter-jdbc-mysql'. "+
+					"But this dependency is not found in your project: %s.", message, prj.Path)
+			}
+			if prj.Metadata.ContainsPropertySpringDatasourcePassword {
+				message = fmt.Sprintf("%s This property should be deleted: "+
+					"'spring.datasource.password'. "+
+					"But this property is found in your project: %s.", message, prj.Path)
+			}
+			continueOption, err := console.Select(ctx, input.ConsoleOptions{
+				Message: fmt.Sprintf("%s Select an option:", message),
+				Options: []string{
+					"Exit azd and fix problem manually",
+					"Continue azd and provision " + database.Display(),
+					"Continue azd but not provision " + database.Display(),
+				},
+			})
+			if err != nil {
+				return false, err
+			}
+
+			switch continueOption {
+			case 0:
+				os.Exit(0)
+			case 1:
+				return true, nil
+			case 2:
+				// remove related database usage
+				var result []appdetect.DatabaseDep
+				for _, db := range prj.DatabaseDeps {
+					if db != database {
+						result = append(result, db)
+					}
+				}
+				prj.DatabaseDeps = result
+				detect.Services[i] = prj
+				// delete database
+				delete(detect.Databases, database)
+				return false, nil
```

Comment:
@haoozhang 

Seems there is some problem about related code in AzureDepEventHubs, could you please fix it in another RP?
Here is the link to related PR: https://github.com/azure-javaee/azure-dev/pull/27/files